src/net.cpp

```diff
@@ -2605,18 +2721,19 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
         bool fFeeler = false;
         std::optional<Network> preferred_net;
 
-        // Determine what type of connection to open. Opening
-        // BLOCK_RELAY connections to addresses from anchors.dat gets the highest
-        // priority. Then we open OUTBOUND_FULL_RELAY priority until we
-        // meet our full-relay capacity. Then we open BLOCK_RELAY connection
-        // until we hit our block-relay-only peer limit.
-        // GetTryNewOutboundPeer() gets set when a stale tip is detected, so we
-        // try opening an additional OUTBOUND_FULL_RELAY connection. If none of
-        // these conditions are met, check to see if it's time to try an extra
-        // block-relay-only peer (to confirm our tip is current, see below) or the next_feeler
-        // timer to decide if we should open a FEELER.
-
-        if (!m_anchors.empty() && (nOutboundBlockRelay < m_max_outbound_block_relay)) {
+        // Determine what type of connection to open, in order of priority:
+        // * PRIVATE_BROADCAST if needed
+        // * BLOCK_RELAY connections to addresses from anchors.dat until we reach m_max_outbound_block_relay
+        // * OUTBOUND_FULL_RELAY until we reach m_max_outbound_full_relay
+        // * BLOCK_RELAY until we reach m_max_outbound_block_relay
+        // * OUTBOUND_FULL_RELAY if GetTryNewOutboundPeer() is true (a stale tip is detected)
+        // * BLOCK_RELAY if it's time to try an extra block-relay-only peer (to confirm our tip is current)
+        // * FEELER if it's time to try a feeler
+        // * else retry the loop (sleep a bit and start from the top of this list)
+
+        if (open_private_broadcast_to.has_value()) {
+            conn_type = ConnectionType::PRIVATE_BROADCAST;
```

Comment:
A few considerations:

* Private broadcast connections are not blocked on the `semOutbound` grant. They come to this code right away.

* Even if there is a high demand for private broadcast connections, we would try to open such ones only half of the time in `CConnman::ThreadOpenConnections()` to avoid such starvation of only opening private broadcast and not others for a long time (see how `MaybePickPrivateBroadcastNetwork()` yields to other connection types if we previously opened a private broadcast one).

* Non-`BLOCK_RELAY` connections keep us up to date as well (with new blocks). I.e blocks could come via inbound, full-outbound, etc.

Given the above, I think the current code does not need to be changed.